snap/squashfs/squashfs.go

```diff
@@ -110,6 +113,11 @@ func (s *Snap) Install(targetPath, mountDir string, opts *snap.InstallOptions) (
 		}
 	}
 
+	if opts.SkipOnDiskSnapValidation && osutil.FileExists(targetPath) {
+		didNothing := true
+		return didNothing, nil
+	}
+
 	// nothing to do, happens on e.g. first-boot when we already
 	// booted with the OS snap but its also in the seed.yaml
 	if s.path == targetPath || osutil.FilesAreEqual(s.path, targetPath) {
```

Comment:
If, as you write, your intent is only to avoid the expensive byte-by-byte comparison, it would seem more readable to me if you put the two code paths in the same if.

Something like:
```
	if s.path == targetPath {
		didNothing := true
		return didNothing, nil
	}
	if osutil.FileExists(targetPath) {
		if opts.SkipOnDiskSnapValidations || osutil.FilesAreEqual(s.path, targetPath) {
			didNothing := true
			return didNothing, nil
		}
	}
```
The conditions can be furtherly combined to minimize the `if` blocks, but IMHO if you leave it like this you could even do without any comments as the code is explicit and obvious.